damnit/gui/table.py

```diff
@@ -627,6 +633,96 @@ def handle_variable_set(self, var_info: dict):
                 self.column_titles[col_ix] = title
                 self.setHorizontalHeaderItem(col_ix, QtGui.QStandardItem(title))
 
+    def handle_processing_running(self, info):
+        processing_id = info['processing_id']
+        self.processing_jobs[processing_id] = info
+        self.update_processing_status(info['proposal'], info['run'])
+        log.debug("Processing running for p%s r%s on %s (%s)",
+                  info['proposal'], info['run'], info['hostname'], processing_id)
+
+    def handle_processing_finished(self, info):
+        processing_id = info['processing_id']
+        info = self.processing_jobs.pop(processing_id, None)
+        if info is not None:
+            self.update_processing_status(info['proposal'], info['run'])
+            log.debug("Processing finished for p%s r%s (%s)",
+                      info['proposal'], info['run'], processing_id)
+
+    def update_processing_status(self, proposal, run):
+        """Show/hide the processing indicator for the given run"""
+        jobs_for_run = [i for i in self.processing_jobs.values()
+                        if i['proposal'] == proposal and i['run'] == run]
+        try:
+            row_ix = self.find_row(proposal, run)
+        except KeyError:
+            if jobs_for_run:
+                row_ix = self.insert_run_row(proposal, run, {}, {}, {})
+            else:
+                return
+
+        runnr_item = self.item(row_ix, 2)
+        if jobs_for_run:
+            runnr_item.setData(f"{run} ⚙️", Qt.ItemDataRole.DisplayRole)
+            if len(jobs_for_run) == 1:
+                info = jobs_for_run[0]
+                msg = f"Processing on {info['hostname']}"
+                if job_id := info['slurm_job_id']:
+                    msg += f" (Slurm job {job_id})"
+                runnr_item.setToolTip(msg)
+            else:
+                runnr_item.setToolTip(f"Processing in {len(jobs_for_run)} jobs")
+        else:
+            runnr_item.setData(f"{run}", Qt.ItemDataRole.DisplayRole)
+            runnr_item.setToolTip("")
+
+    def check_slurm_jobs(self):
```

Comment:
I would recommend putting this in the backend because it'll be necessary for the web interface at some point.

Reply:
Good point, I've refactored this to not be part of the table model class now. Yesterday I was just trying to make this bit work as quickly as possible.